src/react/components/StyleForm.jsx

```diff
@@ -0,0 +1,348 @@
+            const firstTarget = targetString.substr(0, targetString.indexOf('.'));
+            let currentTarget = this.state;
+            currentTarget = this.parseThru(currentTarget, targetString, value);
+
+            this.setState({
+                [firstTarget]: {
+                    ...this.state[firstTarget],
+                    currentTarget
+                }
+            });
+        };
+
+        /**
+         * 
+         * @param {Object} targetObject - target object to be parsed through
+         * @param {String} targetString - target parameter in object provided in full dot notation 
+         * @param {String|Number} value - value to be set
+         */
+        this.parseThru = (targetObject, targetString, value) => {
+            if (typeof targetString === 'string') {
+                return this.parseThru(targetObject, targetString.split('.'), value); // first cycle of recursion converts targetString into array
+            } else if (targetString.length == 1 && value !== undefined) {
+                return targetObject[targetString[0]] = value; // We reach end of the recursion
+            } else if (targetString.length === 0) {
+                return targetObject; // target is already on level 0 so no need for recursion
+            } else {
+                return this.parseThru(targetObject[targetString[0]], targetString.slice(1), value); // recursive call and remove first element from array
+            }
+        }
+    }
+
+    _getCurrentTab (tab) {
+        switch(tab) {
+            case 'point':
+                return (
+                    <PointTab
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        markers={ this.props.markers }
+                        styleSettings={ this.state }
+                    />
+                );
+            case 'line':
+                return (
+                    <LineTab
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        lineIcons={ lineIcons }
+                        styleSettings={ this.state }
+                    />
+                );
+            case 'area':
+                return (
+                    <AreaTab
+                        lineIcons={ lineIcons.lineDash }
+                        areaFills={ areaFills }
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        styleSettings={ this.state }
+                    />
+                );
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * @method _createPreview
+     * @description Compose Preview -component
+     * @returns {React.Component} Preview component with provided style
+     */
+    _createPreview () {
```

Comment:
Each form can init their own preview so it shouldn't be done on the main form level